commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala

```diff
@@ -324,9 +330,14 @@ object ReflectionUtil {
     * it will either return `None` if `s == null`, or `Some` object wrapping the an object of the wrapped type.
     */
   private def buildUnitOrOptionFromString(resultType: Class[_], unitType: Class[_], value: String): Try[Any] = Try {
-    val unit = buildUnitFromString(unitType, value).get
-    if (resultType == classOf[Option[_]]) Option(unit)
-    else unit
+    if (resultType == classOf[Option[_]] && value == SpecialEmptyOrNoneToken) {
```

Comment:
I don't think this change should be necessary.  `buildUnitOrOptionFromString` is only called from `typedValueFromString`, and that will never pass in the `SpecialEmptyOrNoneToken`.